testing/integration/agent_long_test.go

```diff
@@ -0,0 +1,294 @@
+	installPackage.ID = policyUUID
+	installPackage.PolicyID = policyID
+	installPackage.Namespace = "default"
+	installPackage.Name = fmt.Sprintf("%s-long-test-%s", name, policyUUID)
+	installPackage.Vars = map[string]interface{}{}
+
+	runner.T().Logf("Installing %s package....", name)
+	_, err = runner.info.KibanaClient.InstallFleetPackage(ctx, installPackage)
+	require.NoError(runner.T(), err, "error creating fleet package")
+}
+
+func (runner *ExtendedRunner) TestHandleLeak() {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Hour)
+	defer cancel()
+
+	testRuntime := os.Getenv("LONG_TEST_RUNTIME")
+	if testRuntime == "" {
+		testRuntime = "20m"
+	}
+
+	// because we need to separately fetch the PIDs, wait until everything is healthy before we look for running beats
+	require.Eventually(runner.T(), func() bool {
+		allHealthy := true
+		status, err := runner.agentFixture.ExecStatus(context.Background())
+
+		require.NoError(runner.T(), err)
+		for _, comp := range status.Components {
+			runner.T().Logf("component state: %s", comp.Message)
+			if comp.State != int(cproto.State_HEALTHY) {
+				allHealthy = false
+			}
+		}
+		return allHealthy
+	}, time.Minute*3, time.Second*20)
+
+	handles := []processWatcher{}
+
+	regex := regexp.MustCompile(`[\d]+`)
+	status, err := runner.agentFixture.ExecStatus(context.Background())
+	require.NoError(runner.T(), err)
+	// track running beats
+	// the `last 30s` metrics tend to report gauges, which we can't use for calculating a derivative.
+	// so separately fetch the PIDs
+	for _, comp := range status.Components {
+		pidStr := regex.FindString(comp.Message)
+		pid, err := strconv.ParseInt(pidStr, 10, 64)
+		require.NoError(runner.T(), err)
+
+		handle, err := sysinfo.Process(int(pid))
+		require.NoError(runner.T(), err)
+		handlesReg := new(regression.Regression)
+		handlesReg.SetObserved(fmt.Sprintf("%s handle usage", comp.Name))
+		handlesReg.SetVar(0, "time")
+
+		runner.T().Logf("created handle watcher for %s (%d)", comp.Name, pid)
+		handles = append(handles, processWatcher{handle: handle, pid: int(pid), name: comp.Name, regHandles: handlesReg})
+	}
+
+	testDuration, err := time.ParseDuration(testRuntime)
+	require.NoError(runner.T(), err)
+
+	timer := time.NewTimer(testDuration)
+	defer timer.Stop()
+
+	// time to perform a health check
+	ticker := time.NewTicker(time.Second * 10)
+	defer ticker.Stop()
+
+	done := false
+	start := time.Now()
+	for !done {
+		select {
+		case <-timer.C:
+			done = true
+		case <-ticker.C:
+			err := runner.agentFixture.IsHealthy(ctx)
+			require.NoError(runner.T(), err)
+			// for each running process, collect memory and handles
+			for _, handle := range handles {
+
+				ohc, ok := handle.handle.(types.OpenHandleCounter)
+				if ok {
+					handleCount, err := ohc.OpenHandleCount()
+					require.NoError(runner.T(), err)
+					handle.regHandles.Train(regression.DataPoint(float64(handleCount), []float64{time.Since(start).Seconds()}))
```

Comment:
I still think the raw data should be written to a file as a build artifact, otherwise debugging why this failed will be challenging.

Reply:
So, on failure it will print the datapoints to the console, which I figured was fine? At least, it might be more obvious than hiding something in the build artifacts?

Reply:
What format do they have? If it's a file, you can write code against to check the output immediately. If its in the console you are going to have to manually remove the buildkite formatting or write something else to do that first.

If you have sample output it'd help see if this is something that is actually worth addressing or not.

Reply:
So, the regression library we're using prints a debug output like this:
```
    slope_test.go:51: handles|     duration
        17.00|  10.00
        13.00|  20.00
        16.00|  30.00
        13.00|  40.00
        17.00|  50.00
        13.00|  60.00
        16.00|  70.00
        13.00|  80.00
        16.00|  90.00
        14.00|  100.00
        18.00|  110.00
        15.00|  120.00
        12.00|  130.00
        16.00|  140.00
        14.00|  150.00
        18.00|  160.00
        14.00|  170.00
        17.00|  180.00
        15.00|  190.00
        18.00|  200.00
        15.00|  210.00
        18.00|  220.00
        15.00|  230.00
        18.00|  240.00
        14.00|  250.00
        18.00|  260.00
        14.00|  270.00
        17.00|  280.00
        13.00|  290.00
```